docs/design/w3c/spec.md

```diff
@@ -0,0 +1,530 @@
+      presentations, we need to design following things:
+        * how encoding will work for such attributes
+        * how selective disclosure will work on top level attribute itself
+            ```
+              "credentialSubject": {
+                "address": {
+                    "type": "Address",
+                    "city": "Foo",
+                    "street": "Main str."
+                }
+              }
+            ```
+
+#### Data Schemas
+
+W3C [Credential Subject](https://www.w3.org/TR/vc-data-model/#data-schemas) section defines an optional capability to
+include `credentialSchema` property to enforce a specific structure on a given verifiable credential and encoding used
+to map the claims of a verifiable credential to an alternative representation format.
+
+In the context of W3C AnonCreds credentials defined a custom `AnonCredsDefinition` data schema in order to include the
+following information to credential:
+
+In the case of W3C AnonCreds credentials, the `credentialSchema` attribute defines a custom `AnonCredsDefinition`
+schema in order to include the information about Indy related definitions to credential and looks the following:
+
+```
+{
+  ... 
+  "credentialSchema": {
+    "type": "AnonCredsDefinition",
+    "definition": "did:sov:3avoBCqDMFHFaKUHug9s8W:3:CL:13:default",
+    "schema": "did:sov:3avoBCqDMFHFaKUHug9s8W:2:fabername:0.1.0",
+    "encoding": "auto"
+  },
+  ...
+}
+```
+
+* `schema` - id of Indy Schema
+* `definition` - id of Indy Credential Definition
+* `revocation` - id of Indy Revocation Registry
+* `encoding` - attributes encoding algorithm
+    * encoded credential attribute values (binary representation required for doing CL signatures) are not included
+      neither to `credentialSubject` or `signature`
+    * `encoding: auto` implies using the algorithm defined
+      at [Aries RFC 0592 Indy Attachments section](https://github.com/hyperledger/aries-rfcs/tree/main/features/0592-indy-attachments#encoding-claims)
+      to generate encoded values under the hood during the signature generation and proof verification.
+
+#### Issuer
+
+W3C [Issuer](https://www.w3.org/TR/vc-data-model/#issuer) section requires including of `issuer` property to express the
+issuer of a verifiable credential.
+
+In the case of W3C AnonCreds credentials, the `issuer` attribute should be represented as a
+resolvable [DID URL](https://w3c-ccg.github.io/did-resolution/) having either `indy` or `sov` DID method and looks the
+following:
+
+```
+{
+  ... 
+  "issuer": "did:sov:3avoBCqDMFHFaKUHug9s8W",
+  ...
+}
+```
+
+#### Issuance Date
+
+W3C [Issuance Date](https://www.w3.org/TR/vc-data-model/#issuance-date) section requires including of `issuanceDate`
+property to express the date and time when a credential becomes valid.
+
+In the case of W3C AnonCreds credentials, for the `issuanceDate` attribute recommended setting of a random time of the
```

Comment:
Any specific reason for a random time of the day? Couldn't we just say the exact time or set it to 00:00?